core/src/main/java/com/yahoo/oak/InternalOakMap.java

```diff
@@ -325,13 +323,13 @@ V put(K key, V value, OakTransformer<V> transformer) {
         ThreadContext ctx = getThreadContext();
 
         for (int i = 0; i < MAX_RETRIES; i++) {
-            OrderedChunk<K, V> c = findChunk(key); // find orderedChunk matching key
+            BasicChunk<K, V> c = findChunk(key, ctx); // find orderedChunk matching key
```

Comment:
This change and the casting it incurs can be avoided. See comment for `findChunk()`.
Also applicable to other methods.

Reply:
thanks